core/src/main/java/org/infinispan/xsite/irac/DefaultIracManager.java

```diff
@@ -557,23 +578,34 @@ private void onBatchResponse(IracBatchSendResult result, Collection<? extends Ir
       if (log.isTraceEnabled()) {
          log.tracef("[IRAC] Batch completed with %d keys applied. Global result=%s", successfulSent.size(), result);
       }
+      Collection<IracManagerKeyState> doneKeys = new ArrayList<>(successfulSent.size());
       switch (result) {
          case OK:
-            iracExecutor.disableBackOff();
+            backOffHandler.disableBackOff();
+            for (IracManagerKeyState state : successfulSent) {
+               if (sentKeyToAllBackups(state) && state.done()) {
+                  doneKeys.add(state);
+               }
+            }
             break;
          case RETRY:
-            iracExecutor.disableBackOff();
+            backOffHandler.disableBackOff();
```

Comment:
I don't think you need the method `disableBackOff()`. The response collector can invoke `org.infinispan.xsite.XSiteBackup#disableBackOffMode` or `org.infinispan.xsite.XSiteBackup#enableBackOffMode`.

Reply:
I've just kept a call to reset the backoff interval.